proposals/large-ipld-dags.md

```diff
@@ -0,0 +1,117 @@
+
+There is no easy way (e.g., no out-of-band deals) to store large (> sector size) IPLD DAGs while
+preserving the DAG structure.
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+
+* `lotus client deal` should accept an IPLD selector.
+* `lotus client deal` should automatically split large DAGs between multiple sectors.
+* `lotus client retrieve` should support retrieving IPLD selectors (dag subsets).
+
+#### Impact
+_How directly important is the outcome to web3 dev stack product-market fit?_
+
+🔥
+
+At the moment, any tool wishing to support storing IPFS files/directories larger than 32GiB will need to store these IPFS files/directories as "raw blocks", throwing away all the DAG structural information. This will make future retrieval deals for subsets of this data infeasible and will make IPFS interop extremely difficult.
+
+This is only one 🔥 because there are plenty of useful sub-32GiB datasets and non-IPFS datasets.
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+🎯🎯🎯
+
+If we don't solve this now, users will likely store large DAGs any way they can (e.g., as raw
+blocks). We could end up with a lot of unfortunately structured data in Filecoin that's difficult to
+retrieve and work with, especially from IPFS.
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+??
+
+## Project definition
+#### Brief plan of attack
+
+1. Implement selector support in `lotus client deal`. 
+2. Implement selector support in `lotus client retrieve`.
+3. Support automatically splitting large dags into across deals in `lotus client retrieve`. 
+
+#### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+All three of the above commands have been implemented.
+
+NOTE: stopping anywhere along the way will yield a useful result. As long as the first step is finished (selector support for `lotus client deal`), we'll be able to store large structured IPFS data on-chain.
+
+####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+
+1. Developers can easily store large directory trees on Filecoin.
+2. Developers can easily retrieve individual files from large datasets on Filecoin.
+3. Snapshots of English Wikipedia can be stored on Filecoin.
+
+#### Counterpoints &amp; pre-mortem
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+
+The primary risk is that there may be a lack of demand to store large IPFS-formatted datasets in Filecoin. That is, users storing large datasets (> 32GiB) may all be using custom formats and may not care about IPFS files/directories, partial retrieval, etc.
```

Comment:
is this the required path for partial retrievability - or is that a somewhat orthogonal (if related) problem?

Reply:
I'm not sure how the comment relates to the paragraph so I may be misinterpreting it.

Step 2 of the "plan of attack" is required for partial retrieval.